lightning/src/ln/channelmonitor.rs

```diff
@@ -2362,7 +2358,35 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
 		}
 	}
 
-	fn block_connected(&mut self, txn_matched: &[&Transaction], height: u32, block_hash: &Sha256dHash, broadcaster: &BroadcasterInterface, fee_estimator: &FeeEstimator)-> (Vec<(Sha256dHash, Vec<TxOut>)>, Vec<SpendableOutputDescriptor>, Vec<(HTLCSource, Option<PaymentPreimage>, PaymentHash)>) {
+	fn append_htlc_updated(&mut self, mut htlc_updated_infos: Vec<(HTLCSource, Option<PaymentPreimage>, PaymentHash)>) {
+		// ChannelManager will just need to fetch pending_htlcs_updated and pass state backward
+		for htlc in htlc_updated_infos.drain(..) {
+			match self.pending_htlcs_updated.entry(htlc.2) {
+				hash_map::Entry::Occupied(mut e) => {
+					// In case of reorg we may have htlc outputs solved in a different way so
+					// we prefer to keep claims but don't store duplicate updates for a given
+					// (payment_hash, HTLCSource) pair.
```

Comment:
Hmmmm in fact comment isn't that much accurate, if we assume ANTI_REORG_DELAY is safe, the only reason we have this logic is because block-rescan else we shouldn't get the update thanks to onchain_events_waiting_threshold_conf.